### PR TITLE
Mobile responsive navigation menu and invoice list page

### PR DESCRIPTION
**Pull request by Shruti-Apte**
## Notion card
https://www.notion.so/saeloun/Mobile-Responsive-Navigation-Menu-54d119c402ad4d57a01b918b3530a18e
## Summary

 - Designed Mobile responsive navigation menu.
 - Created components for each section in navigation menu.
 - More options section's design is approved by sonam, Akash needs to create design for that section.
 - Mobile responsive Invoice list page 
## Preview

Screenshot 2022-12-06 at 1 46 01 PM

Screenshot 2022-12-06 at 1 52 16 PM

https://www.loom.com/share/32c9de2bb9404ed7b05474f254ffb7b7
## Type of change

- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)

### Checklist:

- [ ] I have manually tested all workflows
- [ ] I have performed a self-review of my own code

### CODE REVIEW

**Review by apoorv1316 — APPROVED** [review]

**Comment by supriya3105** [comment]
@prasanthchaduvula Please review this PR.

File: app/javascript/src/components/Navbar/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+import { Link } from "react-router-dom";
+
+import { Paths } from "constants/index";
+
+const miruLogo = require("../../../../assets/images/PurpleMiruLogoWithText.svg"); //eslint-disable-line
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Navbar/Header.tsx:7]: Any reason why eslint is disabled for this svg import

File: app/javascript/src/components/Navbar/Mobile/Header.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+
+import { Link } from "react-router-dom";
+
+import { Paths } from "constants/index";
+
+const miruLogo = require("../../../../../assets/images/logo.jpg"); //eslint-disable-line
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Navbar/Mobile/Header.tsx:7]: eslint is disabled for this svg import as well. No need to disble this from checks

File: app/javascript/src/components/Navbar/utils.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    path: Paths.PAYMENTS,
+  },
+];
+
+export const activeClassName =
+  "py-3 px-2 md:px-4 flex items-center justify-center md:justify-start text-miru-han-purple-1000 bg-miru-gray-100  border-l-0 md:border-l-8 border-miru-han-purple-1000 font-extrabold";
+
+export const getEmployeeOptions = () =>
+  navEmployeeOptions.map((option, index) => (
+    <li className="items-center hover:bg-miru-gray-100" key={index}>
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Navbar/utils.tsx:69]: Here, the complete `<li>` block is common for both employee and admin options. Please extract the `<li> `to render function and use it inside the employee and admin options

File: app/javascript/src/components/Navbar/Mobile/Navigation.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+  const activeClassName =
+    "flex flex-col items-center justify-center text-miru-han-purple-1000 font-bold text-xs";
+
+  return (
+    <>
+      <ul className="fixed bottom-0 left-0 right-0 flex h-20 justify-between bg-white px-3 shadow-c1">
+        {isAdminUser
+          ? navAdminOptions.slice(0, 4).map((option, index) => (
+              <li
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Navbar/Mobile/Navigation.tsx:21]: Here, the complete `<li>` block is common for both employee and admin options. Please extract the `<li> `to render function and use it inside the employee and admin options

File: app/javascript/src/components/Navbar/Mobile/MoreOptions.tsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    <SidePanel.Header className="w-full">
+      <Header selectedTab="More Options" />
+    </SidePanel.Header>
+    <SidePanel.Body className="flex flex-col" hasFooter={false}>
+      <ul className="h-full w-full pt-14" onClick={() => setVisiblity(false)}>
+        {isAdminUser
+          ? navAdminOptions
+              .slice(4, navAdminOptions.length)
+              .map((option, index) => (
+                <li
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Navbar/Mobile/MoreOptions.tsx:22]: Here, the complete `<li>` block is common for both employee and admin options. Please extract the `<li> `to render function and use it inside the employee and admin options

File: app/javascript/src/components/Navbar/Mobile/MoreOptions.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import { navAdminOptions, navEmployeeOptions } from "../utils";
+
+const MoreOptions = ({ isAdminUser, setVisiblity, setSelectedTab }) => (
+  <SidePanel setFilterVisibilty={setVisiblity}>
+    <SidePanel.Header className="w-full">
+      <Header selectedTab="More Options" />
+    </SidePanel.Header>
+    <SidePanel.Body className="flex flex-col" hasFooter={false}>
+      <ul className="h-full w-full pt-14" onClick={() => setVisiblity(false)}>
+        {isAdminUser
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Navbar/Mobile/MoreOptions.tsx:18]: I think the complete `{isAdminUser}` is common for both `MoreOptions` and `Navigation` components. Extract the logic to another component and reuse it in these both components

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

File: app/javascript/src/components/Invoices/List/MoreOptions.tsx [hunk 7]
```diff
@@ -1,47 +1,173 @@
...
+import { useOutsideClick } from "helpers";
+import {
+  PaperPlaneTiltIcon,
+  DeleteIcon,
+  PrinterIcon,
+  PenIcon,
+  DotsThreeVerticalIcon,
+  DownloadSimpleIcon,
+} from "miruIcons";
+import { Link } from "react-router-dom";
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/Invoices/List/MoreOptions.tsx:12]: @Shruti-Apte Import packages on top
```
import { Link } from "react-router-dom";
```

File: app/javascript/src/components/App.tsx [hunk 8]
```diff
@@ -9,24 +9,52 @@ import UserContext from "context/UserContext";
...
   const isAdminUser = [Roles.ADMIN, Roles.OWNER].includes(companyRole);
+  const [isDesktop, setIsDesktop] = useState<boolean>(window.innerWidth > 1025);
+  const [selectedTab, setSelectedTab] = useState<string>("Time Tracking");
 
   useEffect(() => {
     setAuthHeaders();
     registerIntercepts();
   }, []);
 
+  window.addEventListener("resize", () =>
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/App.tsx:26]: @Shruti-Apte extract the whole view logic to the `DisplayView ` component and make the `App` component clean.

File: app/javascript/src/components/App.tsx [hunk 9]
```diff
@@ -9,24 +9,52 @@ import UserContext from "context/UserContext";
...
+
   return (
     <UserContext.Provider value={{ isAdminUser, user, companyRole }}>
       <BrowserRouter>
         <ToastContainer autoClose={TOASTER_DURATION} />
-        <div className="absolute inset-0 flex h-full w-full">
-          <Navbar isAdminUser={isAdminUser} user={user} />
-          <Main {...props} isAdminUser={isAdminUser} />
-        </div>
+        {getView()}
```
Comment (prasanthchaduvula) [inline app/javascript/src/components/App.tsx:57]: @Shruti-Apte Call `<DisplayView /> ` here

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Review by prasanthchaduvula — APPROVED** [review]

**Comment by Shruti-Apte** [comment]
@apoorv1316 Can you fix the CI tests!

**Pull request merged**